### ISSUE TITLE
Shows previous-Space windows when summoned during a Space transition

### ISSUE THREAD

**Issue by kudryashov13** [lwouis/alt-tab-macos#1254]
Hello, thank you for this app, it's amazing!

I faced with some strange behaviour when using multiple desktops and 'Show windows from Visible Spaces' option at the same time.

I recorded a short video, to show this behavior

https://user-images.githubusercontent.com/57399846/143532276-9ebcf9fb-45b6-477e-88af-f60f2616f178.mov

As you can see on the video I stared from desktop with 2 Google Chrome windows opened, and 1 Mail App window.
Then I switched to the next desktop with VS Code and Spotify window apps. I opened AltTab and it show all windows from previous desktop plus VS Code window. Then I closed and opened AltTab again, now it show correct app windows from current active desktop. I understand that switch desktop and quickly run AltTab not othen use case but it would be nice if there is some solution for this kind of problem for me. I searched through existing Issues but didn't found any where this problem was described.

Steps to reproduce:
1) Create 2 desktops
2) Place different app windows on them
3) Switch between desktops
4) After switching try open Alt-Tab as fast as possible
ER: Shows only apps from current active desktop
AR: Shows apps from previous desktop + current desktop apps

<details>
  <summary>My machine</summary>
  <img src="https://user-images.githubusercontent.com/57399846/143532767-6b84b869-a6d7-4c3f-94fb-651cba5f5ec4.png" width="500" alt="About This Mac">
</details>

<details>
  <summary>AltTab version</summary>
  <img src="https://user-images.githubusercontent.com/57399846/143533741-7bd4dbed-9a18-4ba3-9c85-213c0fd0fe0a.png" width="500" alt="AltTab version">
</details>

<details>
  <summary>AltTab settings</summary>
<img width="500" alt="Screenshot 2021-11-26 at 09 44 30" src="https://user-images.githubusercontent.com/57399846/143533764-246a9d5b-dbe0-4187-9c00-8042e0e7dd3b.png">
<img width="500" alt="Screenshot 2021-11-26 at 09 44 17" src="https://user-images.githubusercontent.com/57399846/143533769-1dea5019-f8a3-4917-af1a-b415450d7f8d.png">
</details>

**Comment by lwouis** [comment]
Hi @kudryashov13 and thanks for this high-quality bug report!

There is already a ticket for this actually, from a year ago: https://github.com/lwouis/alt-tab-macos/issues/689

That being said, your ticket has much better material to explain the issue. The other was a quick ticket I opened to revisit later, but never ended up having time to do. So I'll close it, and let's use yours.

I checked the git logs, and you can see this is not the first time there are edge-cases around this:

![image](https://user-images.githubusercontent.com/106195/143549047-898262c9-5bd7-4fab-a0c7-c8ecf926a163.png)

Just a few examples I'll share so that you know the joy it is to deal with this:

* During the transition, AltTab needs to know which screen is the active screen (to honor `Preferences > Appearance > Show on: [Active screen]` ). Well what do you think the OS API returns when asked for the main screen during a Space transition when it's in the middle?
* What if you hit the keyboard shortcut in the middle of the transition, and hold the `tab` key down? Holding `tab` triggers a key repeating behavior normally. But during a transition, AltTab UI is not visible, so we need to prevent it from moving to the next thumbnails one by one invisibly, so that when the UI finally shows, it's still on the first thumbnail, and *then* it starts to key repeat one by one.

I remember that at some point I changed the implementation on how the AltTab window deals with Space transitions. Spaces are really not properly handled by macOS. There is almost no exposed API to deal with them. Here we want the window to be shown on the new Space as soon as the animation is over.

For instance, here are some APIs that I think influence how the OS deals with the windows during Space transitions. Their documentation doesn't cover the effects properly of course (macOS docs are a joke), but in my tests I found influences when tweaking those:

* [disableSnapshotRestoration](https://developer.apple.com/documentation/appkit/nswindow/1526239-disablesnapshotrestoration)
* [preservesContentDuringLiveResize](https://developer.apple.com/documentation/appkit/nsview/1483795-preservescontentduringliveresize)
* [isFloatingPanel](https://developer.apple.com/documentation/appkit/nspanel/1531901-isfloatingpanel)
* [CollectionBehavior.canJoinAllSpaces](https://developer.apple.com/documentation/appkit/nswindow/collectionbehavior/1419246-canjoinallspaces) -> this one is the most important

To summarize, your issue is valid, and we need to tweak the implementation to do better (if that's actually possible) 👍
